client/value.go

```diff
@@ -79,13 +79,11 @@ func (val FieldValue) Bytes() ([]byte, error) {
 }
 
 func convertImmutable[T any](vals []immutable.Option[T]) []any {
-	var out []any
-	for _, val := range vals {
-		if !val.HasValue() {
-			out = append(out, nil)
-			continue
+	out := make([]any, len(vals))
+	for i := range vals {
+		if vals[i].HasValue() {
+			out[i] = vals[i].Value()
```

Comment:
suggestion: this is a small change with potentially significant impact. I wonder why no tests cover this.
Would be great to add some tests.

Reply:
It is tested, I was mistaken in your PR - I just rewrote it because it was really ugly

Reply:
I don't think this change has any impact other than improving the efficiency of the code. The returned result is the same.